src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,100 @@
+
+/**
+ * Created by banjoe on 5/3/17.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+
+    private EnrollmentService enrollmentService;
+
+    private EnrollmentDto enrollmentDetails;
+    private Student student;
+    private StudentProfile studentProfile;
+    private Set<CourseClass> enrolledClasses;
+    private CourseClass courseClass1;
+    private CourseClass courseClass2;
+
+    @Mock StudentProfileService studentProfileService;
+    @Mock CourseClassService courseClassService;
+    @Mock EnrollmentNotificationService enrollmentNotificationService;
+    @Mock StudyLoadFormatter studyLoadFormatter;
+    @Mock SystemService systemService;
+
+    @Before
+    public void setup() throws Exception {
+        enrollmentService = new EnrollmentService();
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+        enrollmentService.setSystemService(systemService);
+
+        enrollmentDetails = mock(EnrollmentDto.class);
+        student = mock(Student.class);
+        when(enrollmentDetails.getStudent()).thenReturn(student);
+
+        studentProfile = mock(StudentProfile.class);
+        when(enrollmentDetails.getStudentProfile()).thenReturn(studentProfile);
+
+        courseClass1 = mock(CourseClass.class);
+        courseClass2 = mock(CourseClass.class);
+
+        enrolledClasses = new HashSet<>();
+        enrolledClasses.add(courseClass1);
+        enrolledClasses.add(courseClass2);
+        when(enrollmentDetails.getEnrolledClasses()).thenReturn(enrolledClasses);
+    }
+
+    @Test
+    public void processEnrollment_shouldUpdateStudentProfile() throws Exception {
+        enrollmentService.processEnrollment(enrollmentDetails);
+        verify(studentProfileService, times(1)).updateStudentProfile(ArgumentMatchers.eq(studentProfile));
+    }
+
+    @Test
+    public void processEnrollment_shouldEnrollStudentToAClass() throws Exception {
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        verify(courseClassService, times(1)).enrollStudentToClass(ArgumentMatchers.eq(enrollmentDetails.getStudent()), ArgumentMatchers.eq(courseClass1));
+        verify(courseClassService, times(1)).enrollStudentToClass(ArgumentMatchers.eq(enrollmentDetails.getStudent()), ArgumentMatchers.eq(courseClass2));
+    }
+
+    @Test
+    public void processEnrollment_shouldNotifyStudentsEnrollmentDetailsByEmail() throws Exception {
+        enrollmentService.processEnrollment(enrollmentDetails);
+        verify(enrollmentNotificationService, times(1)).emailStudent(ArgumentMatchers.eq(StudentEmailType.ENROLLMENT), ArgumentMatchers.eq(enrollmentDetails));
+    }
+
+    @Test
+    public void processEnrollment_shouldReturnPrintableStudyLoad() throws Exception {
```

Comment:
This is not a concept on its own; the next test case is. The `StudyLoadFormatter`'s duty is to take in the `enrollmentDetails` and transform it into an object that would be acceptable for `SystemService` for it to print later on. In other words, the concept to test is to print the study load; to do so, the `enrollmentDetails` must first be converted to a `Printable` object.

This is a redundant test case.